services/actions/context.go

```diff
@@ -160,3 +164,37 @@ func mergeTwoOutputs(o1, o2 map[string]string) map[string]string {
 	}
 	return ret
 }
+
+func (g *GiteaContext) ToGitHubContext() *model.GithubContext {
+	return &model.GithubContext{
+		Event:            (*g)["event"].(map[string]any),
+		EventPath:        (*g)["event_path"].(string),
+		Workflow:         (*g)["workflow"].(string),
+		RunID:            (*g)["run_id"].(string),
+		RunNumber:        (*g)["run_number"].(string),
+		Actor:            (*g)["actor"].(string),
+		Repository:       (*g)["repository"].(string),
+		EventName:        (*g)["event_name"].(string),
+		Sha:              (*g)["sha"].(string),
+		Ref:              (*g)["ref"].(string),
+		RefName:          (*g)["ref_name"].(string),
+		RefType:          (*g)["ref_type"].(string),
+		HeadRef:          (*g)["head_ref"].(string),
+		BaseRef:          (*g)["base_ref"].(string),
+		Token:            "", // deliberately omitted for security
+		Workspace:        (*g)["workspace"].(string),
+		Action:           (*g)["action"].(string),
+		ActionPath:       (*g)["action_path"].(string),
+		ActionRef:        (*g)["action_ref"].(string),
+		ActionRepository: (*g)["action_repository"].(string),
+		Job:              (*g)["job"].(string),
+		JobName:          "", // not present in GiteaContext
+		RepositoryOwner:  (*g)["repository_owner"].(string),
+		RetentionDays:    (*g)["retention_days"].(string),
+		RunnerPerflog:    "", // not present in GiteaContext
+		RunnerTrackingID: "", // not present in GiteaContext
+		ServerURL:        (*g)["server_url"].(string),
+		APIURL:           (*g)["api_url"].(string),
+		GraphQLURL:       (*g)["graphql_url"].(string),
```

Comment:
Don't do this ........

The failed type-casting will cause panic and shutdown the Gitea instance ......

Reply:
#34508